var/spack/test_repos/spack_repo/builtin_mock/packages/arch_specific_pkg/package.py

```diff
@@ -0,0 +1,20 @@
+# Copyright Spack Project Developers. See COPYRIGHT file for details.
+#
+# SPDX-License-Identifier: (Apache-2.0 OR MIT)
+import platform
+
+from spack_repo.builtin_mock.build_systems.generic import Package
+
+import spack.mirrors.utils
+from spack.package import *
+
+
+class ArchSpecificPkg(Package):
+    if spack.mirrors.utils.evaluate_or_true_if_mirror_all(
+        platform.machine() in ("x86_64", "AMD64")
+    ):
```

Comment:
While the feature is generally meant for "bypassing" platform checks, I think it might make the test more-robust to avoid them:

```
version(1.0, ...) # a version that is always added

if spack.mirrors.utils.evaluate_or_true_if_mirror_all(False):
    version(1.0, ...)
```

I want to avoid the test failing on some system that doesn't fall into any of these categories.